### PR TITLE
mrazadar/PON-208/resubscribe-subscription

### PR DESCRIPTION
**Pull request by mrazadar**
- Subscription stripe-checkout post set timeout to 15 seconds
- improved error handling to fallback-error
- removed fractional zeros with price if fraction is zero according to design
- fixed the i18n warnings and some tests
- added the resubscribe state changes according to design

Screenshot 2023-05-17 at 2 39 03 AM
Screenshot 2023-05-17 at 2 39 22 AM

### CODE REVIEW

File: src/subscription/details/SubscriptionDetails.jsx [hunk 1]
```diff
@@ -77,9 +77,19 @@ export const SubscriptionDetails = () => {
               programUuid,
             }}
           >
-            <SummaryTable price={price} isSubscription />
-            <TotalTable total={totalPrice} isSubscription />
-            {isCurrencyConverted ? <CurrencyDisclaimer currencyDisclaimerSelector={currencyDisclaimerSelector} /> : null}
+            <SubscriptionSummaryTablePrice
```
[the remark this answers is not in the document]
Comment (mrazadar) [reply src/subscription/details/SubscriptionDetails.jsx:80]: Not an issue as discussed with Juliana

File: src/payment/data/utils.js [hunk 2]
```diff
@@ -207,3 +207,22 @@ export const localizedCurrencySelector = () => {
     showAsLocalizedCurrency,
   };
 };
+
+/**
+ * hideFractionZeros
+ * this function will hide fractional zeros if shouldHide is true
+ * and price fractional value is zeros
```
[the remark this answers is not in the document]
Comment (mrazadar) [reply src/payment/data/utils.js]: updated names

File: src/payment/data/utils.js [hunk 3]
```diff
@@ -207,3 +207,22 @@ export const localizedCurrencySelector = () => {
...
+ */
+export const hideFractionZerosProps = ({ price, shouldHide }) => {
+  let fractionDigitsProps = {};
+  if (shouldHide) {
+    const fractionValue = price.toString().split('.')[1];
```
[the remark this answers is not in the document]
Comment (mrazadar) [reply src/payment/data/utils.js:222]: not a bug

File: src/payment/cart/LocalizedPrice.jsx [hunk 4]
```diff
@@ -50,13 +51,15 @@ LocalizedPrice.propTypes = {
...
   showAsLocalizedCurrency: PropTypes.bool,
+  hideFractionZeros: PropTypes.bool,
 };
 
 LocalizedPrice.defaultProps = {
   amount: undefined,
   conversionRate: 1,
   currencyCode: 'USD',
   showAsLocalizedCurrency: false,
+  hideFractionZeros: false,
```
[the remark this answers is not in the document]
Comment (mrazadar) [reply src/payment/cart/LocalizedPrice.jsx]: updated name

File: src/feedback/data/sagas.js [hunk 5]
```diff
@@ -69,14 +69,8 @@ export function* handleSubscriptionErrors(e, clearExistingMessages) {
...
       const error = e.errors[i];
-      /**
-       * * If msg has errorCode and userMessage show it otherwise fallback-error
-       * */
-      if (error.code && error.userMessage) {
-        yield put(addMessage(error.code, error.userMessage, error?.data, error.messageType || MESSAGE_TYPES.ERROR));
-      } else {
-        yield put(addMessage('fallback-error', error.userMessage, error?.data, MESSAGE_TYPES.ERROR));
-      }
+      // there is only fallback-error handling for subscription at this moment.
```
[the remark this answers is not in the document]
Comment (mrazadar) [reply src/feedback/data/sagas.js:72]: will handle in upcoming PR.

**Review by julianajlk — APPROVED** [review]
Thanks for addressing my questions over Slack

**Review by MuhammadMattiullah — APPROVED** [review]

**Review by NawfalAhmed — APPROVED** [review]

**Pull request merged**